proposals/policy_status.md

```diff
@@ -0,0 +1,183 @@
+
+# Definitions
+
+[definitions]: #definitions
+
+- **Policy:** A set of rules in Kyverno that define desired configurations or behaviors for Kubernetes resources.
+- **Webhook:** A mechanism in Kubernetes that allows external services (like Kyverno) to intercept and potentially modify API requests.
+- **Admission Controller:** A type of Kyverno controller that intercepts and processes API requests before they are persisted to the Kubernetes cluster.
+- **Background Controller:** A type of Kyverno controller that processes resources after they are created or updated.
+- **Cache:** An internal data store used by Kyverno to quickly access frequently used information, such as policies.
+- **RBAC (Role-Based Access Control):** A Kubernetes system for controlling who can access and modify resources.
+- **CRD (Custom Resource Definition):** A way to extend Kubernetes with custom resource types.
+- **NotReady:** A condition type that indicates that the specific sub-system is not ready and needs attention.
+
+# Motivation
+
+[motivation]: #motivation
+
+- **Why should we do this?** Currently, diagnosing issues with Kyverno policies can be challenging due to limited status information. This lack of clarity leads to wasted time and effort during troubleshooting.
+- **What use cases does it support?**
+  - Quickly identifying the root cause of policy failures (e.g., webhook misconfiguration, missing permissions).
+  - Automating monitoring and alerting based on policy readiness.
+  - Improving the overall reliability of policy enforcement.
+  - Gaining a better understanding of Kyverno's operational state.
+- **What is the expected outcome?** Clear, actionable status information for each policy will enable users to quickly resolve issues, prevent policy deployment failures, and ensure that policies are enforced as intended. The modification to the `kubectl` output will provide a quick overview of policy readiness at a glance.
+
+# Proposal
+
+[proposal]: #proposal
+
+This proposal introduces new status conditions to the Kyverno Policy CRD. These conditions will track the following:
+
+- `WebhookConfigured`: Indicates whether the webhooks required for the policy are set up correctly. Relevant only for Admission controllers.
+- `CachePresence`: Indicates whether the policy has been successfully loaded into Kyverno's internal cache.
+- `RBACPermissions`: Indicates whether Kyverno has the necessary RBAC permissions to enforce the policy.
+
+Each condition will have a status of `True` or `False`. If a condition's type is `NotReady` and the status is `False`, the policy will fail to apply unless explicitly configured otherwise through `validationFailureActionOverrides`. A condition may not be present if it's not relevant to the policy.
```

Comment:
> If a condition's type is `NotReady` and

What is this referring to....is there a `type`?

Reply:
Good catch. This is a typo. The types are what we defined above.